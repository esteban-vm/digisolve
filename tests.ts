import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
