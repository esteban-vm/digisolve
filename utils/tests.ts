import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
