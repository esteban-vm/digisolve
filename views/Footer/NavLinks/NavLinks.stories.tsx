import type { Meta } from '@storybook/react'
import { styled } from '@/styles'
import NavLinks from './NavLinks'

const Wrapper = styled.div`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ul {
    list-style: none;

    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }
  }
`

export default {
  component: NavLinks,
  title: 'Views/Footer/NavLinks',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper className='row'>
          <div className='col span_1_of_2'>
            <Story {...context} />
          </div>
        </Wrapper>
      )
    },
  ],
} as Meta<typeof NavLinks>

export const NavLinksExample = () => <NavLinks />
