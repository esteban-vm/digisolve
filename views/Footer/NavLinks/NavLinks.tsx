import type { BasicComponent } from '@/types'
import Link from 'next/link'
import { styled } from '@/styles'

export const navLinks = ['About us', 'Blog', 'Help & Support', 'iOS App', 'Android App']

const NavLinksComponent: BasicComponent = (props) => {
  return (
    <ul {...props}>
      {navLinks.map((link, index) => (
        <li key={index}>
          <Link href='#' target='_blank'>
            {link}
          </Link>
        </li>
      ))}
    </ul>
  )
}

const NavLinks = styled(NavLinksComponent)`
  float: left;

  li {
    margin-right: 20px;

    a {
      :hover {
        opacity: 0.8;
      }
    }
  }
`

export default NavLinks
