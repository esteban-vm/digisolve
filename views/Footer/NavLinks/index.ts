export { default, navLinks } from './NavLinks'
