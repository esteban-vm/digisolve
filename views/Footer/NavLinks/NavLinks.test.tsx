import { render, cleanup, screen, create } from '@/tests'
import NavLinks, { navLinks } from './NavLinks'

describe('ðŸ§ª NAV LINKS', () => {
  describe('display tests:', () => {
    it('should display the nav links', () => {
      render(<NavLinks />)
      const parent = screen.getByRole('list')

      const items = screen.getAllByRole('listitem')
      const links = screen.getAllByRole('link')

      expect(navLinks.length).toBe(items.length)
      expect(navLinks.length).toBe(links.length)

      for (let index = 0; index < navLinks.length; index++) {
        const item = items[index]
        const link = links[index]
        const text = navLinks[index]
        expect(link).toHaveTextContent(text)
        expect(item).toContainElement(link)
        expect(parent).toContainElement(link)
      }

      cleanup()
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<NavLinks />)
      expect(tree).toMatchSnapshot()
    })
  })
})
