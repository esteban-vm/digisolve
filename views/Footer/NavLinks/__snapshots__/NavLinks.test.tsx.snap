// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª NAV LINKS style tests: should render with correct styles 1`] = `
.emotion-0 {
  float: left;
}

.emotion-0 li {
  margin-right: 20px;
}

.emotion-0 li a:hover {
  opacity: 0.8;
}

<ul
  className="emotion-0"
>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      About us
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      Blog
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      Help & Support
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      iOS App
    </a>
  </li>
  <li>
    <a
      href="#"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      Android App
    </a>
  </li>
</ul>
`;
